source/ed/data.cpp

```diff
@@ -70,14 +70,17 @@ void Data::build_block_id() {
     /// Two vehicle_journeys with the same block_id vj1 are consecutive if
     /// the last arrival_time of vj1 <= to the departure_time of vj2
     std::sort(vehicle_journeys.begin(), vehicle_journeys.end(),
-            [](const types::VehicleJourney* vj1, const types::VehicleJourney* vj2) {
-            if(vj1->block_id != vj2->block_id) {
-                return vj1->block_id < vj2->block_id;
-            } else {
-                return vj1->stop_time_list.back()->arrival_time <=
-                        vj2->stop_time_list.front()->departure_time;
-            }
+              [](const types::VehicleJourney* vj1, const types::VehicleJourney* vj2) {
+        if(vj1->block_id != vj2->block_id) {
+            return vj1->block_id < vj2->block_id;
+        } else if (vj1->utc_to_local_offset != vj2->utc_to_local_offset) {
+            // we don't want to link the splited vjs
+            return vj1->utc_to_local_offset < vj2->utc_to_local_offset;
+        } else {
+            return vj1->stop_time_list.back()->arrival_time <=
+                    vj2->stop_time_list.front()->departure_time;
```

Comment:
I prefere
```c++
 return vj1->stop_time_list.front()->departure_time <= vj2->stop_time_list.front()->departure_time;
```
as the current version is not a strict weak ordering. The succession is already tested line 100.